kerrgeopy/frequencies.py

```diff
@@ -4,9 +4,26 @@
 """
 from .constants import _standardize_params
 from .constants import *
-from scipy.special import ellipk, ellipe, elliprj, elliprf
+from scipy.special import ellipk, ellipe, elliprj, elliprf, elliprd
 from numpy import sin, cos, sqrt, pi, arcsin, floor, where
 
+def _ellipeinc(phi,m):
+    r"""
+    Incomplete elliptic integral of the second kind defined as :math:`E(\phi,m) = \int_0^{\phi} \sqrt{1-m\sin^2\theta}d\theta`.
+    """
+    # count the number of half periods
+    num_cycles = floor(phi/(pi/2))
+    # map phi to [0,pi/2]
+    phi = abs(arcsin(sin(phi)))
+
+    # makes this work for array input without throwing a division by zero warning
+    c = 1/np.where(phi == 0, np.nan, sin(phi)**2)
+    c = np.where(np.isnan(c),np.inf,c)
+
+    # formula from https://en.wikipedia.org/wiki/Carlson_symmetric_form
+    integral = elliprf(c-1,c-m,c)-1/3*m*elliprd(c-1,c-m,c)
+    return where(num_cycles % 2 == 0, num_cycles*ellipe(m)+integral, (num_cycles+1)*ellipe(m)-integral)
```

Comment:
Is there a way of rewriting this so it does not return an array if the user gives it a float? I know this isn't really a public function, but it is nice to have input and output types remain consistent where possible. See https://stackoverflow.com/questions/29318459/python-function-that-handles-scalar-or-arrays for some comments on how to write functions that take in scalars or arrays